### PR TITLE
Update to spdlog 1.8.2

### PR DESCRIPTION
**Pull request by clalancette**
Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

### CODE REVIEW

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14026/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8847/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11720/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14112/)
* Windows Debug [Build Status](http://ci.ros2.org/job/ci_windows/14115/)

**Review by ahcorde — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -74,7 +74,7 @@ macro(build_spdlog)
   )
 endmacro()
 
-if(NOT spdlog_FOUND OR "${spdlog_VERSION}" VERSION_LESS 1.5.0)
+if(NOT spdlog_FOUND OR "${spdlog_VERSION}" VERSION_LESS 1.8.2)
```
Comment (clalancette) [inline CMakeLists.txt]: I'm honestly not sure if we should do this.  On Windows and macOS, this essentially is a no-op; spdlog never exists there, so we'll always build.  Ubuntu Focal ships with spdlog 1.5.0, so we're currently not building it and relying on the system version instead.  If we make this change, we'll start building it on Ubuntu Focal as well.  That has the upside of being the same version we are using on Windows and macOS, but the downside of not using the system version.  Thoughts?

Comment (clalancette) [reply CMakeLists.txt]: After talking to @cottsay about it, I changed this back to 1.5.0.  That means we'll still use the system package on Ubuntu, which is preferable.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14093/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8911/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11786/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14193/)

**Review by wjwwood — APPROVED** [review]

**Comment by clalancette** [comment]
The failures on macOS and Windows are completely unrelated to this PR.  So I'll go ahead and merge this one, thanks for the reviews.

**Pull request merged**